py_ballisticcalc/munition.py

```diff
@@ -11,78 +11,70 @@
 
     def __post_init__(self):
+        if not self.sight_height:
+            self.sight_height = 0
         if not self.twist:
             self.twist = 0
-        if not self.zero_look_angle:
-            self.zero_look_angle = 0
+        if not self.zero_elevation:
+            self.zero_elevation = 0
 
 
 @dataclass
 class Ammo(TypedUnits):
-    """Creates Ammo and Projectile properties"""
-
-    dm: 'DragModel'
-    length: [float, Distance] = field(default_factory=lambda: Set.Units.length)
+    """
+    :param dm: DragModel for projectile
+    :param mv: Muzzle Velocity
+    :param powder_temp: Baseline temperature that produces the given mv
+    :param temp_modifier: Change in velocity w temperature: fps per °C.
+        Can be computed with .calc_powder_sens().  Only applies if
+        Settings.USE_POWDER_SENSITIVITY = True.
+    """
+    dm: DragModel = field(default=None)
     mv: [float, Velocity] = field(default_factory=lambda: Set.Units.velocity)
+    powder_temp: [float, Temperature] = field(default_factory=lambda: Set.Units.temperature)
     temp_modifier: float = field(default=0)
-    powder_temp: [float, Temperature] = field(default_factory=lambda: Temperature.Celsius(15))
+
+    def __post_init__(self):
+        if not self.powder_temp:
+            self.powder_temp = Temperature.Celsius(15)
 
     def calc_powder_sens(self, other_velocity: [float, Velocity],
                          other_temperature: [float, Temperature]) -> float:
-        """Calculates velocity correction by temperature change
-        :param other_velocity: other velocity
+        """Calculates velocity correction by temperature change; assigns to self.temp_modifier
+        :param other_velocity: other velocity at other_temperature
         :param other_temperature: other temperature
         :return: temperature modifier
         """
-        # (800-792) / (15 - 0) * (15/792) * 100 = 1.01
-        # creates temperature modifier in percent at each 15C
-        v0 = self.mv >> Velocity.MPS
+        v0 = self.mv >> Velocity.FPS
         t0 = self.powder_temp >> Temperature.Celsius
-        v1 = Set.Units.velocity(other_velocity) >> Velocity.MPS
+        v1 = Set.Units.velocity(other_velocity) >> Velocity.FPS
         t1 = Set.Units.temperature(other_temperature) >> Temperature.Celsius
 
-        v_delta = math.fabs(v0 - v1)
-        t_delta = math.fabs(t0 - t1)
-        v_lower = v1 if v1 < v0 else v0
-
-        if v_delta == 0 or t_delta == 0:
-            raise ValueError(
-                "Temperature modifier error, other velocity "
-                "and temperature can't be same as default"
-            )
-
-        self.temp_modifier = v_delta / t_delta * (15 / v_lower)  # * 100
-
+        if t0 == t1:
+            # Can't calculate powder sensitivity without temperature differential.
+            self.temp_modifier = 0.
+        else:
+            self.temp_modifier = (v0 - v1) / (t0 - t1)
```

Comment:
That's not match the original conception, lots of calculators operates a value in `%/15C` measure, so i use it

Reply:
I can't find a definitive source for this question: Should we express temperature sensitivity in deltaV%/deltaT, or in deltaV/deltaT?  Everything I've found so far lists it in the second form – e.g., *change in fps* per *degree change in temperature*.  So something like, +1fps/1degreeF.  Here's one reference: https://www.wlcastleman.com/equip/shoot/t223/index.htm

It sounds like you are saying it should be the first form: *multiple of change in velocity* per *degree change in temperature*.  So something like +0.15%/1degreeF.  And you further say that the preferred terms are in %/15degreesC, so a typical value would be 1%/15degreesC.  We can of course use those terms; I just want to make sure that's exactly what you want?

Reply:
I want to use %/15C cause soldiers in my country uses it common. I know that Applied Ballistics uses fps/degree or mps/degree and that's partially more understandable. But proportional will be more accurate, I tested some projectiles with labradat and got similar values as calculated wia proportional velocities. Also we uses %/15degreesC instead of %/degreeC cause for more accurate data we need a tDelta >= 15C, it gives more valid average powder sensitivity calculations when we trying to get powder_sends by radar data. So using %/15C gets the shooter a hint